tests/python/pytorch/graphbolt/test_feature_fetcher.py

```diff
@@ -1,39 +1,76 @@
-import dgl
-import dgl.graphbolt
-import pytest
+import dgl.graphbolt as gb
+import gb_test_utils
 import torch
+from torchdata.datapipes.iter import Mapper
 
 
-def get_graphbolt_fetch_func():
-    feature_store = {
-        "feature": dgl.graphbolt.TorchBasedFeature(torch.randn(200, 4)),
-        "label": dgl.graphbolt.TorchBasedFeature(torch.randint(0, 10, (200,))),
-    }
+def test_FeatureFetcher_homo():
+    graph = gb_test_utils.rand_csc_graph(20, 0.15)
+    a = torch.randint(0, 10, (graph.num_nodes,))
+    b = torch.randint(0, 10, (graph.num_edges,))
 
-    def fetch_func(data):
-        return feature_store["feature"].read(data), feature_store["label"].read(
-            data
-        )
+    features = {}
+    keys = [("node", None, "a"), ("edge", None, "b")]
+    features[keys[0]] = gb.TorchBasedFeature(a)
+    features[keys[1]] = gb.TorchBasedFeature(b)
+    feature_store = gb.BasicFeatureStore(features)
 
-    return fetch_func
+    itemset = gb.ItemSet(torch.arange(10))
+    minibatch_dp = gb.MinibatchSampler(itemset, batch_size=2)
+    num_layer = 2
+    fanouts = [torch.LongTensor([2]) for _ in range(num_layer)]
+    data_block_converter = Mapper(minibatch_dp, gb_test_utils.to_node_block)
+    sampler_dp = gb.NeighborSampler(data_block_converter, graph, fanouts)
+    keys.append(("node", None, "c"))
+    fetcher_dp = gb.FeatureFetcher(sampler_dp, feature_store, keys)
 
+    assert len(list(fetcher_dp)) == 5
 
-def get_tensor_fetch_func():
-    feature_store = torch.randn(200, 4)
-    label = torch.randint(0, 10, (200,))
 
-    def fetch_func(data):
-        return feature_store[data], label[data]
+def get_hetero_graph():
+    # COO graph:
+    # [0, 0, 1, 1, 2, 2, 3, 3, 4, 4]
+    # [2, 4, 2, 3, 0, 1, 1, 0, 0, 1]
+    # [1, 1, 1, 1, 0, 0, 0, 0, 0] - > edge type.
+    # num_nodes = 5, num_n1 = 2, num_n2 = 3
+    ntypes = {"n1": 0, "n2": 1}
+    etypes = {("n1", "e1", "n2"): 0, ("n2", "e2", "n1"): 1}
```

Comment:
will not the CI fail? if the should e1 be `n1:e1:n2`?

Reply:
No, here it is a tuple not string.